web/src/pages/login.tsx

```diff
@@ -1,23 +1,38 @@
 import { LoginEmailCheck } from "@/components/LoginEmailCheck";
 import { PageSkeleton } from "@/components/njwds-layout/PageSkeleton";
+import { AuthContext } from "@/contexts/authContext";
+import { IsAuthenticated } from "@/lib/auth/AuthContext";
 import { getActiveUser } from "@/lib/auth/sessionHelper";
 import { ROUTES } from "@/lib/domain-logic/routes";
 import { useMountEffect } from "@/lib/utils/helpers";
 import { type GetStaticPropsResult } from "next";
 import { useRouter } from "next/compat/router";
+import { useContext, useEffect } from "react";
 
 import { ReactElement } from "react";
 
 const LoginEmailCheckPage = (): ReactElement => {
+  const { state: authState, dispatch } = useContext(AuthContext);
+
   const router = useRouter();
 
   useMountEffect(() => {
     getActiveUser()
-      .then(() => {
-        router && router.push(ROUTES.dashboard);
+      .then((activeUser) => {
+        dispatch({
+          type: "LOGIN",
+          activeUser: activeUser,
+        });
       })
-      .catch(() => {});
+      .catch(() => (): void => {});
```

Comment:
[dust] Do we want to find a way to make this more readable and to have better diagnostics around this?

Reply:
In the interest of getting this in, gonna defer this one to a later update, but do you have any thoughts on what could go here? I thought about adding a comment to clarify the business logic, something along the lines of:
```ts
.catch(() => (): void => {
     // getActiveUser` throws an error for unauthenticated users.
     // In that case, we don't want to take any action, since 
     // being at the `/login` page makes sense for them.
})
```
But perhaps some conditional logging might be helpful here, too?

Reply:
- Conditional logging, like behind a debugging feature flag
- Naming both functions
- Making the error message checks more explicit by creating explicit subclasses of `Error` and checking for those with differential logging based on error message type

These are all valid approaches and the true answer may be some combination thereof. I know we have some outstanding opportunities to be more explicit with catching errors in the frontend, but I'm open to any possibility to make changes to this code more approachable.